### DISCUSSION TITLE
如何将选出的IP改到官方warp的连接IP上

### DISCUSSION THREAD

**Discussion by Zch0** [XIU2/CloudflareSpeedTest#392] [提问] [answered]
如题

**Comment by XIU2** [comment, marked as answer]
不能，因为 Warp+ 和 网站 区别很多：

1. **节点不同**：网站用的节点 和 Warp+ 用的节点是不一样的，只有个别 IP 段才是分配给 Warp+ 使用的
2. **协议不同**：网站走的是 TCP，而 Warp+ 走的是 UDP，CloudflareST 只能测速 TCP 的流量（及基于 TCP 的 HTTP 协议）
3. **端口不同**：网站用的端口是 80、443 这些，而  Warp+ 用的端口就很多很杂了（可能有几十个端口？

CloudflareST 是为网站优选 IP 而设计的，因此 CloudflareST 不支持优选 Warp+ 节点 IP 。

不过有其他人写了：
https://blog.misaka.rest/2023/03/12/cf-warp-yxip/

> 虽然该文章里的优选工具没有标明作者及开源，但我以前尝鲜 Warp+ 时用过，发现似乎是根据我的 CloudflareST 改的，如果没判断错的话，这工具算是违反了本项目的开源协议 GPL-3.0 license  了（即使用我的开源代码，也必须开源）。。。虽然我懒得追究~

**Reply by puzige** [reply]
是的 他那个脚本是先用shell生成warp的ip然后调用你的可执行文件生成`result.csv`

**Reply by hronro** [reply]
如果这这种调用的话，不算违反 GPLv3。

**Reply by XIU2** [reply]
@hronro 不是调用，可能是拿我源码改的，因为我的 CFST 压根不支持 UDP 测速（不管是 TCPing 还是 HTTPing 都属于 TCP 协议），也不支持多端口测试，自然也不支持 Warp 节点测速（这些服务器只开放了各端口的 UDP 协议），根本不存在调用我的软件来测速 Warp 节点的方式。。。

**Reply by gracelsy** [reply]
刚刚去看了这个blog，真的笑死了

**Reply by CiaoFox** [reply]
这个blog好像被劫持变成挂原主黑料的墙了....看了两眼感受到一点震撼。

**Reply by gracelsy** [reply]
这瓜还是挺香的

**Reply by XIU2** [reply]
+1 我记得有段时间了，挺逗的~

**Reply by a0iwguf09u** [reply]
这个博客变成esuwiki了好恐怖

**Reply by a0iwguf09u** [reply]
https://web.archive.org/web/20240324024044/https://blog.misaka.rest/2023/03/12/cf-warp-yxip/

**Comment by kejunsun** [comment]
最近warp封IP很厉害，有啥好方法可以科学上网吗

**Reply by beck-8** [reply]
机场

**Reply by docker-master** [reply]
CF workers建立trojan节点给很多流量

**Comment by cy920820** [comment]
这是来自邮箱的自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
